DataFormats/L1TGlobal/interface/AXOL1TLScore.h

```diff
@@ -0,0 +1,71 @@
+#ifndef DataFormats_L1TGlobal_AXOL1TLScore_h
+#define DataFormats_L1TGlobal_AXOL1TLScore_h
+
+/**
+* \class AXOL1TLScore
+*
+*
+* Description: L1 micro Global Trigger - Extra NN score emulation information for AXOL1TL.
+* 
+*
+* \author: Melissa Quinnan - UC San Diego
+*
+*
+*/
+
+// system include files
+#include <vector>
+#include <iostream>
+#include <iomanip>
+
+// user include files
+#include "FWCore/Utilities/interface/typedefs.h"
+#include "DataFormats/L1Trigger/interface/L1Candidate.h"
+#include "DataFormats/L1Trigger/interface/BXVector.h"
+#include "DataFormats/L1Trigger/interface/L1TObjComparison.h"
+
+// forward declarations
+
+// namespace l1t {
+class AXOL1TLScore;
+typedef BXVector<AXOL1TLScore> AXOL1TLScoreBxCollection;
+
+typedef l1t::ObjectRef<AXOL1TLScore> AXOL1TLScoreRef;
+typedef l1t::ObjectRefBxCollection<AXOL1TLScore> AXOL1TLScoreRefBxCollection;
+// typedef l1t::ObjectRefPair<AXOL1TLScore> AXOL1TLScoreRefPair;
+// typedef l1t::ObjectRefPairBxCollection<AXOL1TLScore> AXOL1TLScoreRefPairBxCollection; //dont think pair part is needed
+
+// class interface
+class AXOL1TLScore {
+public:
+  /// constructors
+  AXOL1TLScore();  //empty constructor
+
+  AXOL1TLScore(int bxInEvent);
+
+  AXOL1TLScore(int bxInEvent, float score);
+
+  /// destructor
+  virtual ~AXOL1TLScore();
+
+  ///set/get axo score and other simple members
+  void setAXOScore(float score) { axoscore_ = score; }
+  void setbxInEventNr(int bxNr) { m_bxInEvent = bxNr; }
+
+  inline float const& getAXOScore() const { return axoscore_; }
+  inline const int getbxInEventNr() const { return m_bxInEvent; }
```

Comment:
I include it so that I am able to set the score to zero for non bx=0 vertices, because that is what is reflected in firmware.

Reply:
I don't follow. I can find a call to `setbxInEventNr` only in
https://github.com/cms-sw/cmssw/pull/45227/files#diff-b7054a379ef91f6e0ad005651052337e3ced24f9a658ac11a277bf25e1f491c0R497
and there
 - the score can be set to zero based on the value of `iBxInEvent`, and
 - the BX information in the 'output object' is provided via `BXVector` (`AXOL1TLScoreBxCollection`).

So, i do not see why the call to `setbxInEventNr` is needed (and `getbxInEventNr` does not seem to be used anywhere).

Reply:
I added it as a member of the DataFormat because there is a score dependence on iBxInEvent so I thought it could be useful to include that information, and because I was modeling the score DataFormat after the GlobalAlgBlk DataFormat, which is set up similarly. https://github.com/cms-sw/cmssw/blob/master/DataFormats/L1TGlobal/interface/GlobalAlgBlk.h#L57. I suppose if you feel strongly it can be removed, but I personally think it doesn't do any harm.

Reply:
> because I was modeling the score DataFormat after the GlobalAlgBlk DataFormat

I see (from a quick look, `GlobalAlgBlk::m_bxInEvent` looks pretty unnecessary, too).

> I suppose if you feel strongly it can be removed

No preference from my side (not that i get a vote). I guess L1T-sw has the final say on the L1T data formats. I just wanted to understand if this data member was really needed.

Reply:
Ok, maybe lets leave it as is for now if that's ok